controllers/applicationsController/getAllApplications.js

```diff
@@ -17,20 +16,14 @@ const getAllApplications = async (req, res, next) => {
   try {
     const { organizationId } = req.token;
     if (!mongoose.Types.ObjectId.isValid(organizationId)) {
-      return next(new CustomError(400, "Invalid OrganizationID"));
-    }
-
-    const organization = await Organizations.find({ _id: organizationId });
-    if (!organization) {
-      return next(new CustomError(400, "Invalid organization"));
+      return next(new CustomError(422, "Invalid OrganizationID"));
```

Comment:
Looking at this again maybe we should change this to invalid organization token instead?

Reply:
We don't even need to know if the organization exists or not in this controller for it to perform its function. That job should be done by another.

Reply:
What other function or controller checks this? An organization may have been removed and the organization token used before it expires, we may end up allowing such invalid requests through without this check

Reply:
Can't the authentication middleware that runs check this before the controller method is called? That way, the check is done once for all applications endpoints and the code respects the DRY principle. If we add the check here then we will need to add the check in every application and admin endpoint.

Reply:
You are right. I guess we may need to use mongo hooks and middleware to ensure foreign relationships/references are maintained. So if e.g an org is deleted it's respective apps will be deleted as well